packages/react/src/primitives/ScrollView/ScrollView.tsx

```diff
@@ -11,23 +11,42 @@ import {
 } from '../types';
 import { View } from '../View';
 import { primitiveWithForwardRef } from '../utils/primitiveWithForwardRef';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
 
 const ScrollViewPrimitive: Primitive<ScrollViewProps, 'div'> = (
-  { children, className, orientation, ...rest },
-  ref
-) => (
-  <View
-    className={classNames(
-      ComponentClassName.ScrollView,
-      classNameModifier(ComponentClassName.ScrollView, orientation),
-      className
-    )}
-    ref={ref}
-    {...rest}
-  >
-    {children}
-  </View>
-);
+  { children, className, orientation, autoScroll, ...rest },
+  externalRef
+) => {
+  const internalRef = React.useRef<HTMLDivElement | null>(null);
+  const composedRefs = useComposeRefsCallback<HTMLDivElement | null>({
+    externalRef,
+    internalRef,
+  });
+
+  React.useLayoutEffect(() => {
```

Comment:
What's the reasoning for `useLayoutEffect` rather than `useEffect` and do we need to support SSR?

Reply:
TBH I dunno, changed it to useEffect and it works fine